### PR TITLE
fix(rome_js_analyze): useExhaustiveDependencies support function syntax, or React.use* hooks and more.

### PR DESCRIPTION
**Pull request by nissy-dev**
## Summary

This PR fixed most of #4330

useExhaustiveDependencies rule now works correctly 

- when the first argument of hooks is a named function
- inside an export default function
- for React.use* hooks

## Test Plan

I added some snapshot tests.

## Changelog

- [x] The PR requires a changelog line

## Documentation

- [ ] The PR requires documentation
- [ ] I will create a new PR to update the documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,10 @@
 ### Linter
 
 - Fix a crash in the `NoParameterAssign` rule that occurred when there was a bogus binding. [#4323](https://github.com/rome/tools/issues/4323)
+- Fix useExhaustiveDependencies rule in the following cases [#4330](https://github.com/rome/tools/issues/4330)
```
Comment (ematipico) [inline CHANGELOG.md:39]:
```suggestion
- Fix `useExhaustiveDependencies` rule in the following cases [#4330](https://github.com/rome/tools/issues/4330)
```

File: crates/rome_js_analyze/tests/specs/nursery/useExhaustiveDependencies/missingDependenciesInvalid.js.snap [hunk 2]
```diff
@@ -89,6 +89,44 @@ const MyComponent8 = React.memo(({ a }) => {
...
+  let a = 1;
+  useEffect(async function inner() {
+      console.log(a);
+  });
+}
+
+// React.useXXX case
+function MyComponent13() {
+  let a = 1;
+  React.useEffect(() => {
```
Comment (ematipico) [inline crates/rome_js_analyze/tests/specs/nursery/useExhaustiveDependencies/missingDependenciesInvalid.js.snap:125]: We should make sure that:
- `React` is imported `import React from "react"`, I believe we already have some API for that;
- `React` is not created here, e.g.
```js
const React = { useEffect() {} }
React.useEffect();
```

Can we create some test against these cases?

Comment (nissy-dev) [reply crates/rome_js_analyze/tests/specs/nursery/useExhaustiveDependencies/missingDependenciesInvalid.js.snap:125]: Thank you for your review!

> Can we create some test against these cases?

Yes. l'll try to update some codes to pass your test cases.

Comment (nissy-dev) [reply crates/rome_js_analyze/tests/specs/nursery/useExhaustiveDependencies/missingDependenciesInvalid.js.snap:128]: I fixed in [b644eb6]!(https://github.com/rome/tools/pull/4571/commits/b644eb6e2118e4cd95401c8350c1af1db4175372)

**Review by ematipico — APPROVED** [review]

**Pull request merged**